### PR TITLE
Blob

### PR DESCRIPTION
**Pull request by ESaracay**
Adds image functionality to comments.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -31,15 +31,18 @@
...
 public class DataServlet extends HttpServlet {
   public class Comment {
-    private String timeCreated, user, content;
+    private String timeCreated, user, content, key;
 
-    public Comment(String user, String content, String time) {
+    public Comment(String user, String content, String time, String key) {
       this.user = user;
       this.content = content;
+      this.key = key;
```
Comment (tylerhou) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Key is too ambiguous, maybe `imageKey` or `imageBlobKey`?

File: portfolio/src/main/java/com/google/sps/servlets/NewCommentDataServlet.java [hunk 2]
```diff
@@ -14,15 +14,32 @@
...
 /**
  * Runs when the form for a new comment is submitted. The comment
  * is then stored permenantly with DatastoreService.
  */
 @WebServlet("/new-comment")
 public class NewCommentDataServlet extends HttpServlet {
+
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String blobKey= getBlobKey(request);
```
Comment (tylerhou) [inline portfolio/src/main/java/com/google/sps/servlets/NewCommentDataServlet.java]: Format

File: portfolio/src/main/java/com/google/sps/servlets/NewCommentDataServlet.java [hunk 3]
```diff
@@ -63,4 +81,50 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
   private boolean verified(String user, String content) {
     return !user.isEmpty() && !content.isEmpty();
   }
+
+
+  private String getBlobKey(HttpServletRequest request){
```
Comment (tylerhou) [inline portfolio/src/main/java/com/google/sps/servlets/NewCommentDataServlet.java]: This method is named "getBlobKey", which makes me think that it has no [side effects](https://en.wikipedia.org/wiki/Side_effect_(computer_science)), but it actually uploads and deletes images.

Usually getter methods shouldn't do anything besides getting, so I think you can find a better name for this.

File: portfolio/src/main/java/com/google/sps/servlets/NewCommentDataServlet.java [hunk 4]
```diff
@@ -63,4 +81,50 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
+
+    if (myKeys == null || myKeys.isEmpty()) {
+      return null;
+    }
+
+    BlobKey blobKey = myKeys.get(0);
+
+    BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
+    if (blobInfo.getSize() == 0) {
+      myBlobService.delete(blobKey);
```
Comment (tylerhou) [inline portfolio/src/main/java/com/google/sps/servlets/NewCommentDataServlet.java]: Leave a comment explaining why we have to delete? Why would this code run?

File: portfolio/src/main/java/com/google/sps/servlets/ServeImageServlet.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+import java.lang.Thread;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.List;
+import java.util.Map;
+//BlobStore
```
Comment (tylerhou) [inline portfolio/src/main/java/com/google/sps/servlets/ServeImageServlet.java]: Reminder to delete

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -209,9 +216,17 @@ async function grabComments() {
...
-    fetch('/delete-comments', {
+  if (confirm("Are you sure you want to delete all comments")) {
+    fetch("/delete-comments", {
       method: 'POST'
     }).then(() => setTimeout(grabComments, 1000));
   }
 }
+
+async function grabBlobURL(){
+    blobURL = await fetch("/blobstore-upload-url").then((response) => {return response.text();});
```
Comment (tylerhou) [inline portfolio/src/main/webapp/script.js]: `const blobURL`

also, format.

File: portfolio/src/main/webapp/script.js [hunk 7]
```diff
@@ -209,9 +216,17 @@ async function grabComments() {
...
+    const myForm = document.getElementById("comment-form");
+    myForm.action = blobURL;
+   // myForm.classList.remove('myclass');
+    console.log(blobURL);
```
Comment (tylerhou) [inline portfolio/src/main/webapp/script.js]: Reminder to remove.

File: portfolio/src/main/webapp/script.js [hunk 8]
```diff
@@ -104,9 +111,15 @@ async function grabComments() {
...
-  if (confirm('Are you sure you want to delete all comments')) {
-    fetch('/delete-comments', {
+  if (confirm("Are you sure you want to delete all comments")) {
+    fetch("/delete-comments", {
       method: 'POST'
     }).then(() => setTimeout(grabComments, 1000));
   }
 }
+
+async function grabBlobURL(){
```
Comment (tylerhou) [inline portfolio/src/main/webapp/script.js]: Function name implies that it doesn't have side effects, but it does.

**Review by tylerhou — APPROVED** [review]

File: portfolio/src/main/webapp/script.js (hunk 8 above)
Comment (ESaracay) [reply portfolio/src/main/webapp/script.js]: done.

**Pull request merged**